### PR TITLE
OCPBUGS-5461: Allow NO_PROXY to contain IPs

### PR DESCRIPTION
**Pull request by derekhiggins**
Update the validation on NO_PROXY to allow IP addresses to be include.

### CODE REVIEW

File: pkg/types/validation/installconfig.go [hunk 1]
```diff
@@ -770,9 +770,10 @@ func validateProxy(p *types.Proxy, c *types.InstallConfig, fldPath *field.Path)
 			v = strings.TrimSpace(v)
 			errDomain := validate.NoProxyDomainName(v)
 			_, _, errCIDR := net.ParseCIDR(v)
-			if errDomain != nil && errCIDR != nil {
+			ip := net.ParseIP(v)
+			if errDomain != nil && errCIDR != nil && ip == nil{
```
Comment (barbacbd) [inline pkg/types/validation/installconfig.go:774]: shouldn't this be `or` comparison as the error message states?

Comment (derekhiggins) [reply pkg/types/validation/installconfig.go:774]: No, we enter here (and fail validation) if all 3 conditions report an error, i.e. the value being tested isn't any of the 3 supported types.

**Pull request merged**